### ISSUE TITLE
Deepspeed for the forget model

### ISSUE THREAD

**Issue by sina-nabavi** [yihuaihong/ConceptVectors#4]
Hi,
Thank you for this interesting project!

This is not an issue with the code, but a question about its functionality. 

I believe the current version of `forget.py` does not utilize deepspeed and model sharding. I noticed the deepspeed config given to `TrainingArguments` is commented. My question is have you used or considered deepspeed for this part of your project in any of your experiments? I am trying to run `forget.py` on a cluster of 40GB A100s and get OOM errors with the code as-is.   

Thanks!

**Comment by yihuaihong** [comment]
Hi, thank you for your interest and question!

In this project’s unlearning fine-tuning, we didn’t use multi-GPU training — training was done on single GPU each time (80GB A100). Therefore, I commented out the DeepSpeed code. If you need to train on multiple GPUs, you’ll likely need to modify the training code, especially the loss calculation part, which may require adjustments.

Thanks!

**Comment by sina-nabavi** [comment]
Thanks so much for the prompt response and clarification. Will do!
